### PR TITLE
8345609: [C1] LIR Operations with one input should be implemented as LIR_Op1

### PR DESCRIPTION
**Pull request by TheRealMDoerr**
Change `lir_sqrt`, `lir_abs`, `lir_neg`, `lir_f2hf`, `lir_hf2f` to use `LIR_Op1`. Extend `LIR_Op1` to support one temp register operand. Also see JBS issue.

Remove `lir_tan` and `lir_log10` which are unused (dead and incomplete code). They should have been removed with or after https://github.com/openjdk/jdk/commit/ad79a5ae65d24861ead3ae96cf148c8bc0f02736 and the corresponding changes on other platforms.

The removal of the unnecessary float constant loads improves performance:
make run-test TEST="micro:Fp16ConversionBenchmark" MICRO="VM_OPTIONS=-XX:TieredStopAtLevel=1"

Power 10 without patch:
```
Benchmark                                     (size)   Mode  Cnt      Score    Error   Units
Fp16ConversionBenchmark.floatToFloat16          2048  thrpt   15    247.064 ±  0.189  ops/ms
```

Power 10 with patch:
```
Benchmark                                     (size)   Mode  Cnt      Score    Error   Units
Fp16ConversionBenchmark.floatToFloat16          2048  thrpt   15    308.372 ±  0.432  ops/ms
```

x64 machine without patch:
```
Benchmark                                     (size)   Mode  Cnt      Score    Error   Units
Fp16ConversionBenchmark.floatToFloat16          2048  thrpt   15     384.565 ±    3.758  ops/ms
```

x64 machine with patch:
```
Benchmark                                     (size)   Mode  Cnt      Score    Error   Units
Fp16ConversionBenchmark.floatToFloat16          2048  thrpt   15     406.121 ±    3.228  ops/ms
```

Testing: tier1-4 on x64 (Windows, linux, MacOS), aarch64 (linux, MacOS), ppc64 (linux, AIX)

---------
### Progress
- [x] Change must be properly reviewed (1 review required, with at least 1 [Reviewer](https://openjdk.org/bylaws#reviewer))
- [x] Change must not contain extraneous whitespace
- [x] Commit message must refer to an issue

### Issue
 * [JDK-8345609](https://bugs.openjdk.org/browse/JDK-8345609): [C1] LIR Operations with one input should be implemented as LIR_Op1 (**Enhancement** - P4)

### Reviewers
 * [Richard Reingruber](https://openjdk.org/census#rrich) (@reinrich - **Reviewer**)
 * [Goetz Lindenmaier](https://openjdk.org/census#goetz) (@GoeLin - **Reviewer**)

### Reviewing
<details><summary>Using <code>git</code></summary>

Checkout this PR locally: \
`$ git fetch https://git.openjdk.org/jdk.git pull/22582/head:pull/22582` \
`$ git checkout pull/22582`

Update a local copy of the PR: \
`$ git checkout pull/22582` \
`$ git pull https://git.openjdk.org/jdk.git pull/22582/head`

</details>
<details><summary>Using Skara CLI tools</summary>

Checkout this PR locally: \
`$ git pr checkout 22582`

View PR using the GUI difftool: \
`$ git pr show -t 22582`

</details>
<details><summary>Using diff file</summary>

Download this PR as a diff file: \
<a href="https://git.openjdk.org/jdk/pull/22582.diff">https://git.openjdk.org/jdk/pull/22582.diff</a>

</details>
<details><summary>Using Webrev</summary>

[Link to Webrev Comment](https://git.openjdk.org/jdk/pull/22582#issuecomment-2520929787)
</details>

### CODE REVIEW

File: src/hotspot/share/c1/c1_LIR.cpp [hunk 1]
```diff
@@ -483,6 +489,7 @@ void LIR_OpVisitState::visit(LIR_Op* op) {
 
       assert(op1->_info != nullptr, "");  do_info(op1->_info);
       if (op1->_opr->is_valid())       do_temp(op1->_opr); // safepoints on SPARC need temporary register
+      assert(op1->_tmp->is_illegal(), "not used");
```
Comment (TheRealMDoerr) [inline src/hotspot/share/c1/c1_LIR.cpp:492]: I think safepoints should ideally be implemented as LIR_Op0. SPARC support is removed. We could file a new RFE for that.

**Review by reinrich — APPROVED** [review]
Nice cleanup 👍
A few files need copyright update // especially c1_LinearScan_x86.hpp ;)
Cheers, Richard.

**Comment by TheRealMDoerr** [comment]
Thanks for the review! Copyright headers are updated.

**Review by reinrich — APPROVED** [review]

**Comment by reinrich** [comment]
You might even see an effect in the `Fp16ConversionBenchmark` if you stop compilation at tier 1.

**Comment by TheRealMDoerr** [comment]
> You might even see an effect in the `Fp16ConversionBenchmark` if you stop compilation at tier 1.

Yes. Added benchmark results to the description.

File: src/hotspot/share/c1/c1_LIR.cpp (hunk 1 above)
Comment (GoeLin) [reply src/hotspot/share/c1/c1_LIR.cpp:492]: Well, a grep showed there are more leftovers of sparc and solaris in hotspot, but not a matter of this change.

**Review by GoeLin — APPROVED** [review]
LGTM

**Comment by TheRealMDoerr** [comment]
Thanks for the review!
/integrate

**Comment by dholmes-ora** [comment]
We have a large number of failures in our tier 2 (so far) CI after this was integrated, all with issues in the JIT.

There is a crash in LIR_Assembler::negate

#  assert(regs[i] != regs[j]) failed: regs[0] and regs[1] are both: xmm2

There are FP failures of different kinds

assertEquals expected: 1.401298464324817E-45 but was: 1.727233711018889E-77

RuntimeException: pow(+Infinity, 0.5), expected: Infinity, actual: Infinity

Sorry but this change will be backed out.

**Pull request closed without merging**